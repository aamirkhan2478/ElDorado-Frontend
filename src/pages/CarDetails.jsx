import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { useParams, useNavigate } from 'react-router-dom';
import { car, getCar } from '../redux/Home/home';

const CarDetails = ({ open }) => {
  const { id } = useParams();
  const carDetails = useSelector(car);
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleReserve = () => {
    navigate('/booking');
  };

  useEffect(() => {
    dispatch(getCar(id));
  }, []);
  return (
    <>
      <div className="mt-10 flex items-center smax:flex-col gap-3">
        <img
          className="w-[1000px] min-w-[270px] rounded-md self-stretch object-cover"
          src={carDetails.image}
          alt={carDetails.name}
        />
        <div className="flex flex-col gap-4 self-stretch">
          <h1 className="text-2xl font-semibold self-end justify-self-start font-osans">
            {carDetails.name}
          </h1>

          <p className="font-bold bg-gray-400 text-center px-7">
            {carDetails.model}
          </p>

          <p className=" bg-white text-black text-center font-bold px-7">
            $
            {carDetails.daily_price}
            {' '}
            /day
          </p>
          <p className={`text-base max-w-[800px] ${open && 'text-xs'}`}>{carDetails.description}</p>
          <button onClick={handleReserve} type="button">
            <span className="text-black font-bold bg-amber-500 border px-4 hover:bg-white hover:border-amber-500 hover:text-amber-500 py-3 rounded-sm">
              Reserve
            </span>
          </button>
        </div>
      </div>
    </>
  );
};

CarDetails.propTypes = {
  open: PropTypes.bool.isRequired,
};

export default CarDetails;
