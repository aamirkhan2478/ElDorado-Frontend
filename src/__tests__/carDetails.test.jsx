import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/configureStore';
import '@testing-library/jest-dom/';
import CarDetails from '../pages/CarDetails';

describe('Car detail page', () => {
  test('Should render the proper details page for a car', () => {
    const carDetail = render(
      <Provider store={store}>
        <Router>
          <CarDetails />
        </Router>
      </Provider>,
    );
    expect(carDetail).toMatchSnapshot();
  });
});
