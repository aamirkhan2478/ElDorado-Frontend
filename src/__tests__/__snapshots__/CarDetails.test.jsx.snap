// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car detail page Should render the proper details page for a car 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="overflow-auto mx-auto px-4 py-5 bg-gray-200 rounded"
      >
        <div
          class="md:flex"
        >
          <div
            class="mr-4"
          >
            <h1
              class="text-4xl text-black font-bold self-end justify-self-start font-osans pl-2 mb-12 border-l-4 border-amber-500"
            />
            <img
              alt="Modern building architecture"
              class="w-[1000px] min-w-[270px] rounded-lg object-cover self-stretch"
            />
          </div>
          <div
            class="flex flex-col"
          >
            <div
              class="mx-auto mt-2 text-2xl text-black font-bold self-end justify-self-start font-osans bg-amber-300 px-6 py-2 max-w-fit rounded"
            >
              $
              <sub
                class="font-features subs font-normal"
              >
                /per day
              </sub>
            </div>
            <p
              class="text-xs text-black font-light text-center"
            >
              Included Taxes and Checkup
            </p>
            <div
              class="text-black bg-white border border-gray-200 p-3 my-2 drop-shadow"
            >
              <h2
                class="text-xl border-l-2 border-amber-500 font-bold p-1 mb-4 bg-amber-50"
              >
                SPECIFICATIONS
              </h2>
              <p
                class=""
              >
                <span
                  class="font-semibold "
                >
                  Make:
                </span>
                 
              </p>
              <p
                class="text-gray "
              >
                <span
                  class="font-semibold "
                >
                  Model:
                </span>
                 
              </p>
              <p
                class="text-gray text-base max-w-[420px] true"
              >
                <span
                  class="font-semibold"
                >
                  Description:
                </span>
                 
              </p>
            </div>
            <button
              class="my-3 mx-auto self-center"
              type="button"
            >
              <span
                class="text-black font-bold bg-amber-500 border px-4  hover:bg-amber-600 py-3 rounded-sm my-2"
              >
                Reserve
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="overflow-auto mx-auto px-4 py-5 bg-gray-200 rounded"
    >
      <div
        class="md:flex"
      >
        <div
          class="mr-4"
        >
          <h1
            class="text-4xl text-black font-bold self-end justify-self-start font-osans pl-2 mb-12 border-l-4 border-amber-500"
          />
          <img
            alt="Modern building architecture"
            class="w-[1000px] min-w-[270px] rounded-lg object-cover self-stretch"
          />
        </div>
        <div
          class="flex flex-col"
        >
          <div
            class="mx-auto mt-2 text-2xl text-black font-bold self-end justify-self-start font-osans bg-amber-300 px-6 py-2 max-w-fit rounded"
          >
            $
            <sub
              class="font-features subs font-normal"
            >
              /per day
            </sub>
          </div>
          <p
            class="text-xs text-black font-light text-center"
          >
            Included Taxes and Checkup
          </p>
          <div
            class="text-black bg-white border border-gray-200 p-3 my-2 drop-shadow"
          >
            <h2
              class="text-xl border-l-2 border-amber-500 font-bold p-1 mb-4 bg-amber-50"
            >
              SPECIFICATIONS
            </h2>
            <p
              class=""
            >
              <span
                class="font-semibold "
              >
                Make:
              </span>
               
            </p>
            <p
              class="text-gray "
            >
              <span
                class="font-semibold "
              >
                Model:
              </span>
               
            </p>
            <p
              class="text-gray text-base max-w-[420px] true"
            >
              <span
                class="font-semibold"
              >
                Description:
              </span>
               
            </p>
          </div>
          <button
            class="my-3 mx-auto self-center"
            type="button"
          >
            <span
              class="text-black font-bold bg-amber-500 border px-4  hover:bg-amber-600 py-3 rounded-sm my-2"
            >
              Reserve
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
